File: 0981/Original-Submission/Src/ex5_PDE/LTfun2.m
function U = LTfun2(X,Y,S)
%{
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                EXAMPLE 5

        Application of Talbot's method to solve
        the following PDE problem

                u_t  = u_xx  + u_yy,            0 < x,y < 1,   t>0
                u(x,y,0+) = x*(x-1) + y*(y-1)
                u(0,y,t) = u(1,y,t) = 4*t + y*(y-1)
                u(x,0,t) = u(x,1,t) = 4*t + x*(x-1)

        The analytical solution is

                u(x,y,t) = 4*t + x*(x-1) + y*(y-1)

        and Laplace Transform is

                          4    x*(x-1) + y*(y-1)
                U(x,s) = --- + -----------------
                         s^2           s

        where s = 0 is a double pole.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%}

%   U(h,k) = U(X(h),Y(h),S(k))
    U = zeros(numel(X),numel(S));
	for k=1:numel(S)
		U(:,k) = 4/S(k)^2 + ( X.*(X-1) + Y.*(Y-1) )/S(k);
	end
end

File: 0981/Original-Submission/Src/ex3b_BVP/LTfun2.c
/**     EXAMPLE 3b

    Application of Talbot's method to solve
    the following PDE problem

        u_t (x,t) = u_xx (x,t),		0 < x < L, t > 0
		  u(x,0+) = x*(x-1)
		  u(0,t)  = 2*t
		  u(L,t)  = 2*t + L*(L-1)

    in [0,L]x[T0,T1].

    The analytical solution is

            u(x,t) = x*(x-1) + 2*t

    and Laplace Transform is

                     x*(x-1)    2
            U(x,s) = ------- + ---
                        s      s^2

    where s = 0 is a double pole.
*/

#include <complex.h>
#include <math.h>


double complex LTfun2(double x, double complex s)
/* Laplace Transform function U(x,s) */
{
    return 2.0/(s*s) + x*(x-1)/s;
}
